src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationOptions.cs

```diff
@@ -166,7 +171,35 @@ public AzureAppConfigurationOptions UseFeatureFlags(Action<FeatureFlagOptions> c
             {
                 throw new InvalidOperationException($"Please select feature flags by either the {nameof(options.Select)} method or by setting the {nameof(options.Label)} property, not both.");
             }
-            
+
+            IsFeatureManagementV2SchemaEnabled = options.EnableFeatureManagementV2SchemaPreview;
+
+            if (IsFeatureManagementV2SchemaEnabled)
+            {
+                if (_adapters.Any(a => a is FeatureFlagV1KeyValueAdapter))
+                {
+                    throw new InvalidOperationException($"Please select either Feature Management V1 schema or V2 schema, not both.");
```

Comment:
Can we be more explicit about what's wrong. For example, something like "Opted in to Feature Management V2 schema, but one or more feature flags defined in V1 schema were detected. Use xxx to specify Feature Management schema opt-in preference".